### PR TITLE
chore: adds worker service to stack

### PR DESCRIPTION
**Pull request by xar-tol**
adds worker service stack file
delays integration to next PR due to size of files

Addresses #2550

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/workloads/partials/cf/subscribe.yml [hunk 1]
```diff
@@ -47,6 +47,15 @@ EventsKMSKey:
             - "kms:ReEncrypt*"
             - "kms:GenerateDataKey*"
           Resource: '*'
+        - Sid: "Allow Role Encrypt/Decrypt"
```
Comment (Lou1415926) [inline templates/workloads/partials/cf/subscribe.yml]: nit:
```suggestion
        - Sid: "Allow task role encrypt/decrypt"
```

**Review by Lou1415926 — APPROVED** [review]
LGTM! Great works!

File: internal/pkg/deploy/cloudformation/stack/worker_svc.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+type WorkerService struct {
+	*ecsWkld
+	manifest *manifest.WorkerService
+
+	allowedTopics []string
+	parser        workerSvcReadParser
+}
+
+// NewWorkerService creates a new WorkerService stack from a manifest file.
+func NewWorkerService(mft *manifest.WorkerService, env, app string, rc RuntimeConfig, at []string) (*WorkerService, error) {
```
Comment (bvtujo) [inline internal/pkg/deploy/cloudformation/stack/worker_svc.go]:
```suggestion
func NewWorkerService(mft *manifest.WorkerService, env, app string, rc RuntimeConfig, allowedTopics []string) (*WorkerService, error) {
```

Can we make this input name more descriptive? Will help readability in other packages.

File: templates/workloads/partials/cf/envvars.yml [hunk 3]
```diff
@@ -4,8 +4,10 @@
 Environment:
 - Name: COPILOT_APPLICATION_NAME
   Value: !Sub '${AppName}'
+{{- if not .Subscribe}}
```
Comment (bvtujo) [inline templates/workloads/partials/cf/envvars.yml]: We should leave this in as an environment variable; this variable is for customers to use to construct queries to *other* services in the environment. 

For example:
```
# From inside worker service
curl db.$COPILOT_SERVICE_DISCOVERY_ENDPOINT:80/inventory/remove/item?=dogbowl&count?=2
```

**Review by bvtujo — APPROVED** [review]

**Comment by bvtujo** [comment]
This looks super good. Just a couple nits and ship it. (Feel free to remove the label when done)

**Pull request merged**